examples/inspection/plot_model_inspection_cross_validation.py

```diff
@@ -0,0 +1,222 @@
+"""
+==========================================================
+Inspect and analyze a linear model within cross-validation
+==========================================================
+
+Evaluating a predictive linear model involves :ref:`cross-validation
+<cross_validation>`. This example:
+
+* details how to interpret the results from a cross-validation framework;
+* highlights how to inspect the internals of a model when using the
+  cross-validation framework.
+"""
+
+print(__doc__)
+import sklearn
+
+sklearn.set_config(display="diagram")
+
+# Authors: Guillaume Lemaitre <g.lemaitre58@gmail.com>
+# License: BSD 3 clause
+
+# %%
+# Dataset
+# -------
+#
+# We will use the :ref:`california_housing_dataset` where the goal is to
+# predict the average house value in a neighborhood. From the start, we will
+# split our data into two sets: a set that we will use for all our experiments
+# and a set that we will leave out for further confirmation.
+
+from sklearn.datasets import fetch_california_housing
+from sklearn.model_selection import train_test_split
+
+X, y = fetch_california_housing(as_frame=True, return_X_y=True)
+
+X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.1, random_state=0)
+
+# %%
+# Predictive model
+# ----------------
+#
+# In this example, we will use a linear model which should be a good baseline:
+# a :ref:`ridge model <ridge_regression>`. A ridge model enforces a L2 penalty
+# on the coefficients. Thus, the penalty parameter `alpha` has to be tuned.
+# More importantly, this parameter needs to be tuned for our specific problem:
+# tuning on another dataset does not ensure an optimal parameter value for the
+# current dataset.
+#
+# Here, we use the class :class:`~sklearn.linear_model.RidgeCV` that can tune
+# `alpha` by cross-validation when calling `fit`.
+#
+# We also add a preprocessing stage to :ref:`standardize
+# <preprocessing_scaler>` the data such that the regularization strength is
+# applied homogeneously on each coefficient.
+import numpy as np
+from sklearn.linear_model import RidgeCV
+from sklearn.pipeline import make_pipeline
+from sklearn.preprocessing import StandardScaler
+
+alphas = np.logspace(-1, 2.5, num=50)
+model = make_pipeline(StandardScaler(), RidgeCV(alphas=alphas))
+
+# %%
+# `model` is a machine learning pipeline made of the preprocessing stage
+# followed by a ridge regressor that will perform an internal cross-validation
+# at `fit` to select the best parameter `alpha`. The candidates for the
+# parameter `alpha` are given by the variable `alphas`.
+#
+# Cross-validation framework
+# --------------------------
+#
+# Before putting such a predictive model into production, one needs to evaluate
+# the performance of the model to have an idea of what to expect in production.
```

Comment:
Depending on your use case, this might be by far not enough for production readiness (or it might be overkill).

Reply:
do you have a proposal to improve the statement

Reply:
Just replace "in production". For example:
```suggestion
# Before applying such a predictive model on new data, as is often done in production, it is a good practice and
# generally advisable to perform an unbiased evaluation of the performance of the model.
```